seahorn/lib/sea_ipc_helper.c

```diff
@@ -55,14 +55,10 @@ sea_channel_connect(struct ipc_port_context *parent_ctx,
 }
 
 /*
- * constant variable of ipc_port_context
+ * directly return a port context given a connect handler
  */
-const static struct ipc_port_context ctx = {
-      .ops = {.on_connect = sea_channel_connect},
-  };
-
 struct ipc_port_context* create_port_context(){
-    struct ipc_port_context* pctx = malloc(sizeof(struct ipc_port_context));
-    * pctx = ctx;
+  struct ipc_port_context* pctx = malloc(sizeof(struct ipc_port_context));
```

Comment:
I do not understand the motivation neither for original nor changed code. Why not return pointer to the global ctx object?

what is this functionality? is that local to a trustlet or is it our library?

Reply:
Because I check the `sea_channel_connect` must be a static function. https://github.com/seahorn/verifyTrusty/blob/87ff11e1db904485b168c882a2b2ec68323fba6e/seahorn/lib/sea_ipc_helper.c#L49
Otherwise, we hit an unhandled instruction when we use `sea_channel_connect`.
```
Warning: unhandled instruction:   %_476 = call %struct.ipc_channel_context* %_475(%struct.ipc_port_context* nonnull %_418, %struct.uuid* nonnull %_0, i32 %_452) #10, !dbg !1421 @ _474 in main
```

Reply:
the warning shows that our devirt might not be configured correctly. I would just return pointer to global. 

original code was copying global into locally allocated memory. that is not so good. 

does not it also create a memory leak? how is malloced memory deallocated?

Reply:
Yes, I agree with your doubts. Just did a new push by using a return pointer.